Transfer of the smart contract notion docs (#7)
aragon/builders-portal: docs/core/02-how-to-guides/01-plugin-development/03-publication/index.md

---
title: Publication
---

## Publishing Your Plugin

In the following sections, we guide you through the publication process of your plugin.

> After you finished the implementation and setup contract of your new plugin, you now want to publish it on the Aragon Marketplace.

Reviewer: ```suggestion
After you finished the implementation and setup contract of your new plugin, you now want to publish it within Aragon's UI.
```

Author: 👍 Hey, would be good if you can maintain these changes if these change in the future

Author: I wrote:

> After you finished the implementation and setup contract of your new plugin, you now want to publish it on Aragon's repository.

> Alternatively, you modified an already published implementation its setup contract to

Reviewer: ```suggestion
Alternatively, you may have modified an already-published implementation and want its `PluginSetup` contract to:
```

- fix bugs or vulnerabilities,
- add new features, or

> - make breaking changes
>   that you now want to publish under a new version number.

Reviewer: ```suggestion
- make breaking changes that you now want to publish under a new version number.
```
